src/commandHandlers/roles.ts

```diff
@@ -1,35 +1,83 @@
   const roles = await getRoles();
-  const rolesList = roles
-    .filter((r) => !r.name.includes('everyone')) // Filter the default role everyone has access to
-    .map((discordRole) => {
-      const role = Object.values(config.ROLE).find(
-        (r) => r.name === discordRole.name
-      );
-      const roleDescription = role ? `- ${role.description}` : '';
-
-      return `\n• ${discordRole.toString()} ${roleDescription}`;
-    });
-
-  embed.setTitle('Available Roles')
-    .setDescription(`Here is the list of roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!
-        ${rolesList}
-        Example of usage:
-        ^iam add javascript`);
+  const specificRole = arg[1];
 
+  if (specificRole) {
+    if (arg[1] === '-a' || arg[1] === 'all') {
+      const roleInfo = getRoleLists(roles, (r) => r.name.includes('everyone'));
+      embed.setTitle('Available Roles')
+        .setDescription(`Here is the list of all the roles on this server. You can assign almost any role to yourself. Some of the roles are admin only or given to you via a condition!\n
+        ${roleInfo[0].toString()}
+        ${roleInfo[1].toString()}\n
+        Assigning this role to yourself:
+        \`^iam add javascript\``);
+    } else {
+      // Search the roles list for the argument, and
+      // if the role is not found, send an Error Embed
+      const role = arg[1];
+      const roleInfo = getRoleLists(roles, (r) => !r.name.includes(role));
+      const roleText = [].concat(roleInfo[0], roleInfo[1]).toString();
+      if (roleText.length === 0) {
+        embed
+          .setColor(config.COLORS.alerts)
+          .setTitle('Role Listing (error)')
+          .setDescription(`**${role}** \nThat role does not exist`)
+          .addField('Usage', usage);
+        console.log("ERROR- The queried role can't be found");
+      } else {
+        embed.setTitle(`${role} Role`).setDescription(`${roleText}
+          \nAssigning this role to yourself:
+              \`^iam add ${role}\``);
+      }
+    }
+  } else {
+    const roleInfo = getRoleLists(
+      roles,
+      (r) => !selfAssignableRoles.includes(r.name)
+    );
+    embed.setTitle('Assignable Roles')
+      .setDescription(`Here is the list of all self-assignable roles on this server. You can assign the following roles to yourself:\n
+      ${roleInfo[0].toString()}
+      ${roleInfo[1].toString()}\n
+      Assigning roles to yourself:
+      \`^iam add javascript\``);
+  }
   return embed;
 };
 
+// Returns two lists, one with all the role names and the other with role names followed by their description.
+function getRoleLists(
+  roles: Collection<string, Role>,
+  isValidRole: (role: Role) => boolean
+): [string[], string[]] {
+  const describedRoles: string[] = [];
+  const rolesList: string[] = [];
+  roles.forEach((role) => {
+    if (isValidRole(role)) {
```

Comment:
Maybe we should change the naming of this parameter. It's a bit weird that we `return` if the role is valid (it should be the opposite😅). How about `isNotValidRole` @Vyvy-vi ?

Or we could do `!isValidRole(role)` and some callers need to change, for example `r.name.includes('everyone')` should change to `!r.name.includes('everyone')`

Reply:
yes I was confused for the first few seconds because of the name from the function and was like "why is this doing the opposite 🤔 " :joy: